cgroup/cgroup.go

```diff
@@ -0,0 +1,77 @@
+package cgroup
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/godbus/dbus/v5"
+	"github.com/godbus/dbus/v5/introspect"
+	"github.com/godbus/dbus/v5/prop"
+
+	logging "github.com/home-assistant/os-agent/utils/log"
+)
+
+const (
+	objectPath            = "/io/homeassistant/os/CGroup"
+	ifaceName             = "io.homeassistant.os.CGroup"
+	CGroupFSDockerDevices = "/sys/fs/cgroup/devices/docker/"
+)
+
+type cgroup struct {
+	conn *dbus.Conn
+}
+
+func (d cgroup) UpdateDevicesAllowed(containerID string, permission string) (bool, *dbus.Error) {
+	allowedFile := CGroupFSDockerDevices + containerID + "/devices.allow"
```

Comment:
Kinda smells like path traversal problems could hit us here.

It seems that golang has nothing built-in :( https://github.com/golang/go/issues/20126

But this seems to help: https://github.com/cyphar/filepath-securejoin